controllers/chat/initChatFromPost.js

```diff
@@ -48,147 +175,72 @@ const initChatFromPost = async (req, res) => {
-        anon_user_info_emoji_name: emoji.name
+    /// START
+    const createdChannel = await newChannel.create();
+
+    // get 100 messages
+    const channelFilters = {cid: 'messaging:' + newChannel.id};
+    const messageFilters = {created_at: {$lte: new Date()}};
+    const messageHistory = await client.search(channelFilters, messageFilters, {
+      sort: [{updated_at: -1}],
+      limit: 100
+    });
+    const {betterChannelMember, betterChannelMemberObject, updatedChannel} =
+      await BetterSocialCore.chat.updateBetterChannelMembers(newChannel, createdChannel, true, {
+        channel_type: channel_type
       });
-    }
-    //end add anon info to channel
-
-    try {
-      if (!newChannel?.data?.name) {
-        let channelType = CHANNEL_TYPE.CHAT;
-        let anonDetailInfo = {};
-        if (userModel.is_anonymous) {
-          channelType = CHANNEL_TYPE.ANONYMOUS;
-          anonDetailInfo = {
-            anon_user_info_color_code:
-              newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_code,
-            anon_user_info_color_name:
-              newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_name,
-            anon_user_info_emoji_code:
-              newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_code,
-            anon_user_info_emoji_name:
-              newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_name
-          };
-        }
-
-        if (targetUserModel.is_anonymous && !userModel.is_anonymous) {
-          anonDetailInfo = {
-            anon_user_info_color_code:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code,
-            anon_user_info_color_name:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name,
-            anon_user_info_emoji_code:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code,
-            anon_user_info_emoji_name:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name
-          };
-        }
-
-        await newChannel.updatePartial({
-          set: {
-            channel_type: channelType,
-            name: [userModel?.username, targetUserModel?.username].join(', '),
-            ...anonDetailInfo,
-            better_channel_member: newStateMemberWithAnonInfo
-          }
-        });
-      }
-    } catch (e) {
-      console.log(e);
-    }
-
-    const channelState = await newChannel.watch();
 
     const response = {
-      ...channelState,
-      better_channel_members: Object.values(channelState.channel.better_channel_member),
-      better_channel_members_object: channelState.channel.better_channel_member,
-      messageHistories: []
+      ...updatedChannel,
+      better_channel_members: betterChannelMember,
+      better_channel_members_object: betterChannelMemberObject,
+      messageHistories: messageHistory.results
     };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function is creating a new channel if it doesn't exist. However, there's no error handling for the case when `newChannel.create()` fails. This could lead to unhandled promise rejections.
```diff
    const createdChannel = await newChannel.create();
+    } catch (error) {
+      console.error(`Failed to create channel: ${error}`);
+      throw error;
+    }

    // get 100 messages
    const channelFilters = {cid: 'messaging:' + newChannel.id};
    const messageFilters = {created_at: {$lte: new Date()}};
    const messageHistory = await client.search(channelFilters, messageFilters, {
      sort: [{updated_at: -1}],
      limit: 100
    });
    const {betterChannelMember, betterChannelMemberObject, updatedChannel} =
      await BetterSocialCore.chat.updateBetterChannelMembers(newChannel, createdChannel, true, {
        channel_type: channel_type
       });
 
     const response = {
      ...updatedChannel,
      better_channel_members: betterChannelMember,
      better_channel_members_object: betterChannelMemberObject,
      messageHistories: messageHistory.results
     };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->